tinkerpop/cypher-gremlin-server-client/src/main/java/org/opencypher/gremlin/client/CypherGremlinClient.java

```diff
@@ -195,4 +195,14 @@ default CypherResultSet submit(String cypher, Map<String, ?> parameters) {
      * @return Cypher-style results
      */
     CompletableFuture<CypherResultSet> submitAsync(String cypher, Map<String, ?> parameters);
+
+    /**
+     * Submits a Cypher query asynchronously.
+     *
+     * @param cypher     query text
+     * @param aliases    aliases for graph variables on the server
+     * @param parameters query parameters
+     * @return Cypher-style results
+     */
+    CompletableFuture<CypherResultSet> submitAsync(String cypher, Map<String,String> aliases, Map<String, ?> parameters);
```

Comment:
The new argument is not in line with Gremlin client and duplicates the `alias` functionality of the backing driver. I think there should either be a similar API to alias a `CypherGremlinClient` or you should alias the driver instead and make a new `CypherGremlinClient` based on that.